### PR TITLE
Project Chatbot - Elin Segelöv and Saralie Bognandi

### PR DESCRIPTION
**Pull request by ElinSegelov**
Pull request for Week 1 project

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -1,7 +1,18 @@
 // Variables that point to selected DOM elements
```
Comment (juliaostedt) [inline code/script.js:1]: Suggest to add ``` document.addEventListener("DOMContentLoaded", () => { } ``` around the whole code per Daniel's instruction

File: code/script.js [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
 const chat = document.getElementById('chat');
+const inputWrapper = document.querySelector(".input-wrapper");
+const entireScreen = document.querySelector("#body-id");
+const submitButton = document.querySelector(".send-btn");
+const inputField = document.querySelector("#name-input");
 
 // If you need any global variables that you can use across different functions, declare them here:
+let heroName;
+let heroHp = 2
```
Comment (juliaostedt) [inline code/script.js]: The code comment re heroHp seem to be on the wrong line. Suggestion to specify the comment and explain what HP stands for.

File: code/script.js [hunk 3]
```diff
@@ -36,16 +47,414 @@ const showMessage = (message, sender) => {
 // Starts here
 const greeting = () => {
   // here we call the function showMessage, that we declared earlier with the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, What's your name?", 'bot');
-  // Just to check it out, change 'bot' to 'user' here 👆
+  showMessage("Hi friend 👋, what is your name?", 'bot');
+
 }
 
+setTimeout(greeting, 1);
```
Comment (juliaostedt) [inline code/script.js]: Note that the "time to wait" is very short (1 millisecond), suggest to remove function or extend the milliseconds.

File: code/script.js [hunk 4]
```diff
@@ -36,16 +47,414 @@ const showMessage = (message, sender) => {
...
+
+/* This is the first reply from the bot */
+const nameReply = () => {
+  showMessage(`Please, ${heroName}, help me save my puppy 🐶. Here are some items you may need to succeed on this rescue mission. Choose wisely! 🧐`, "bot");
+  setTimeout(itemSelection, 1000);
+}
+
+
+// List to select items 
+const itemSelection = () => {
```
Comment (juliaostedt) [inline code/script.js:71]: Great and clear function naming!

File: code/script.js [hunk 5]
```diff
@@ -36,16 +47,414 @@ const showMessage = (message, sender) => {
...
+  inputWrapper.innerHTML = `
+    <select id="item-one">
+      <option value="" selected disabled>👇 Select an item</option>
+      <option value="umbrella">Umbrella ☂️</option>
+      <option value="boat">Inflatable boat ⛵️</option>
+      <option value="water-bottle">Bottle of water🥤</option>
+      <option value="life-jacket">Life jacket 🦺</option>
+      <option value="helmet">Helmet ⛑</option>
+      <option value="rope">Rope 🪢</option>
```
Comment (juliaostedt) [inline code/script.js:80]: FYI - rope/sheild/helmet emojis not showing properly on chrome for windows/pc.

File: code/script.js [hunk 6]
```diff
@@ -36,16 +47,414 @@ const showMessage = (message, sender) => {
...
+const pathChoice = () => {
+  let pathSelect = document.querySelectorAll(".path-button");
+  pathSelect.forEach(button => button.addEventListener("click", function (event) {
+    event.preventDefault();
+    inputWrapper.innerHTML = "";
+    console.log(this.id)
+    showMessage(`I will go through the ${this.innerHTML}`, "user");
+   
+    // These are if-statment within an if-statement. Depending on which path the Hero chooses and what it has in it's backpack/inventory different conditions are triggered. 
+    // For one road, one item is perfect, one item makes you loose 1 hp, the rest will give game over.
```
Comment (juliaostedt) [inline code/script.js:132]: Great explaination!

File: code/script.js [hunk 7]
```diff
@@ -36,16 +47,414 @@ const showMessage = (message, sender) => {
...
+    let itemList1 = document.querySelector("#item-one");
+    let itemList2 = document.querySelector("#item-two");
+    let selectItemBtn = document.querySelector("#select-item-btn");
+    selectItemBtn.addEventListener("click", function (event) {
+      event.preventDefault();
+      // the value of the selected option in respective list is assign to variable item1 and item2.
+      item1 = itemList1.options[itemList1.selectedIndex].value;
+      item2 = itemList2.options[itemList2.selectedIndex].value;
+      htmlItem1 = itemList1.options[itemList1.selectedIndex].innerHTML;
+      htmlItem2 = itemList2.options[itemList2.selectedIndex].innerHTML;
```
Comment (juliaostedt) [inline code/script.js:104]: Suggestion to add comment to clarify difference between "item2" and "htmlItem2"

File: code/script.js [hunk 8]
```diff
@@ -36,16 +47,414 @@ const showMessage = (message, sender) => {
...
+        }, 1500);
+        setTimeout(botIntersection2, 5000);
+      }
+      else {
+        console.log("GAME OVER")
+        setTimeout( () => { 
+          showMessage(`The sun burns 🔥 your skin and you sweat every last drop of water 🥵. You die! ☠️`,`bot`); 
+        }, 1500);
+        setTimeout(gameOver, 6000);
+      }
```
Comment (juliaostedt) [inline code/script.js:155]: Super clear code for the conditionals, easy to follow. Great indentations.

File: code/script.js [hunk 9]
```diff
@@ -36,16 +47,414 @@ const showMessage = (message, sender) => {
...
+    showMessage(`Your injuries are too severe. You die! ☠️`, 'bot');
+        showMessage(`Your attemt fails and the dragon 🐲  eats you alive. Why didn't you choose something else to help you on this dangerous mission? You die! ☠️`,`bot`); 
+        }, 1500);
+        setTimeout(gameOver, 8000);
+      }
+    } 
+  }))
+  }
+
+const conclusion = () => {
```
Comment (juliaostedt) [inline code/script.js:345]: Suggest to add comment to describe where in the story the conclusion shows.

File: code/script.js [hunk 10]
```diff
@@ -36,16 +47,414 @@ const showMessage = (message, sender) => {
...
+    </div>
+    <button class="replay-btn">Play again</button>
+    <p> This interactive story was made by Elin Segelöv and Saralie Bognandi.</p>
+    </div>
+    `
+    let retryButton = document.querySelector(".replay-btn");
+    retryButton.addEventListener("click", function (event) {
+      event.preventDefault();
+      location.reload();
+    })
```
Comment (juliaostedt) [inline code/script.js:402]: Nice feature!

File: code/script.js [hunk 11]
```diff
@@ -36,16 +47,414 @@ const showMessage = (message, sender) => {
...
+const deathBy0Hp = () => {
+  if(heroHp === 0) {
+    showMessage(`Your injuries are too severe. You die! ☠️`, 'bot');
+    setTimeout(gameOver, 3000);
+  }
+}
+
+//victory screen function
+const endScreen = () => {
+  entireScreen.innerHTML=`
```
Comment (juliaostedt) [inline code/script.js:386]: Cool with the entire screen changing!

**Review by juliaostedt — APPROVED** [review]
Overall really nice good that works super well! Impressed with all the different paths and storyline. Code was easy to follow and clear namings. Could have added a few more explainatory comments throughout the code.

**Review by miadahlgren — APPROVED** [review]
Julia and I made the code review together so the comments from her is from the both of us. :) great work this week! :)

File: code/script.js (hunk 9 above)
Comment (ElinSegelov) [reply code/script.js:345]: I don't understand this comment. Could you elaborate?

**Pull request closed without merging**